### ISSUE TITLE
Export fields not shown if ModelField of different name defined in ModelResource

### ISSUE THREAD

**Issue by ibowditch** [django-import-export/django-import-export#1893]
**Describe the bug**
If I define a field in a ModelResource such as memb =  fields.Field(column_name='Member', attribute='memb__rfsID'), where neither the field.attribute nor the field.column_name is the same as the field name, it is excluded from the exported list, even if it remains selected in the list of fields in the export form

**To Reproduce**
1. Define a field in a ModelResource with a column_name and attribute different from the field name.
2. Export the model with the new field selected
3. Confirm that the new field is excluded from the export

**Versions (please complete the following information):**
 - Django Import Export: 4.1.0
 - Python 3.8.19
 - Django 3.2.25

**Expected behavior**
All fields that remain selected in the export form should appear in the output.

**Screenshots**

**Additional context**
I patched this in my Resource models, by overriding import_export.resources.Resource._get_enabled_export_fields and replacing the following code:

            return [
                field
                for field in export_fields
                if field.attribute in fields_ or field.column_name in fields_
            ]

with

            return [ 
                self.fields[field_name] 
                for field_name in fields_ 
                if field_name in self.fields and self.fields[field_name] in export_fields 
           ]

This works for me, but not sure if it is a general solution.

**Comment by matthewhegarty** [comment]
Hi
Thanks for reporting.  I confirmed that this is a regression from v3.

If I declare the following in v3 it exports the column correctly as "Author Name":

```python
class EBookResource(ModelResource):
    auteur_name = Field(attribute='author__name', column_name='Author Name')
```

**Comment by ibowditch** [comment]
Hi Matt

I just updated from 4.1.0 to 4.1.1, and Issue #1893 is still present there.

The problem is that _get_enabled_export_fields is returning a list of Fields, rather than a list of Field names. 

I can't see a way of re-opening. Should I submit another report?

Cheers,
Ian

**Comment by matthewhegarty** [comment]
Can you try with the main branch?  This fix has been merged but not released yet.  It should be released in v4.2 which will hopefully be in next couple of months.

**Comment by ibowditch** [comment]
Ok - thanks. No rush, and I'm happy to wait for 4.2.

Thanks for your work on this very useful library

Cheers,
Ian
